python/ray/train/v2/_internal/execution/worker_group/worker_group.py

```diff
@@ -435,6 +435,12 @@ def _clear_state(self):
         self._worker_group_state = None
         self._world_rank_to_ongoing_poll = {}
 
+    def abort(self):
+        """Abort the worker group."""
+        self._assert_active()
+        for callback in self._callbacks:
+            callback.before_worker_group_abort(self._worker_group_context)
```

Comment:
Should we add a call to force shut down the worker group during this step?
So that the actors get explicitly killed by us, rather than rely on being cleaned up by garbage collection?

Reply:
In which case, should `before_worker_group_shutdown` get called still?

One alternative would be to have `abort` just call `shutdown(abort=True)`, and pass `abort=True` down to the callback hooks `before_worker_group_shutdown(self, abort=True)`.

It's a little unintuitive for `WorkerGroup.abort` to basically be a noop.

Reply:
As per our discussion offline, adding a TODO comment explaining the hanging risk for now.